src/PositionManager.sol

```diff
@@ -217,9 +237,35 @@ contract PositionManager is
         if (nativeBalance > 0) recipient.safeTransferETH(nativeBalance);
     }
 
+    /// @param params an encoding of Currency, address
+    function _sweepERC20To(bytes calldata params) internal {
+        (Currency currency, address to) = abi.decode(params, (Currency, address));
+        uint256 tokenBalance = ERC20(Currency.unwrap(currency)).balanceOf(address(this));
+        if (tokenBalance > 0) currency.transfer(to, tokenBalance);
+    }
+
     // implementation of abstract function DeltaResolver._pay
-    function _pay(Currency token, address payer, uint256 amount) internal override {
-        // TODO this should use Permit2
-        ERC20(Currency.unwrap(token)).safeTransferFrom(payer, address(poolManager), amount);
+    function _pay(Currency currency, address payer, uint256 amount) internal override {
+        if (payer == address(this)) {
+            // TODO: This transfer no eth check. This is guaranteed to not be eth.
```

Comment:
> This is guaranteed to not be eth.

say why! also i dont understand what the todo is?

Reply:
Because by the time we call _pay from inside the DeltaResolver settle function we have already checked if its eth. eth follows a diff path in settle